app/models/technology.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class Technology < ApplicationRecord
+  has_many :company_technologies, dependent: :destroy
+  has_many :companies, through: :company_technologies
+
+  validates :name, presence: true, uniqueness: true
+  validates :background_color,
+    format: { with: /\A#(?:[0-9a-fA-F]{3}){1,2}\z/, message: "must be a valid hex color code" },
```

Comment:
Can we extract the regex into a constant?

```ruby
class Technology < ApplicationRecord
  HEX_COLOR_REGEX = /\A#(?:[0-9a-fA-F]{3}){1,2}\z/.freeze
  
  # ...
  
  validates :background_color,
    format: { with: HEX_COLOR_REGEX, ... }
```

We can move it to a custom validator once/if we need it in another model.